portindex2json/portindex2json.tcl

```diff
@@ -2,7 +2,7 @@
 # a list of JSON objects.
 # Written by Joshua Root <jmr@macports.org>, 2017
 # Requires: tclsh with the tcllib 'json' package.
-# Usage: tclsh portindex2json.tcl path/to/PortIndex
+# Usage: tclsh portindex2json.tcl path/to/PortIndex commit-hash(optional)
```

Comment:
I would prefer if this was a named argument. Maybe something like
```
portindex2json.tcl /path/to/PortIndex -id <shasum>
```
also valid when used as
```
portindex2json.tcl -id <shasum> /path/to/PortIndex
```
(I'm not sure what the best name would be; `id` is one option, but it could also be called `shasum` or `sha1` or `commit` or whatever.)

Another option would be to provide an even more flexible setup. Say that at some point in the future you will also want to add a timestamp. If we add something like
```
portindex2json.tcl ... -info id=<shasum> -info timestamp=20190701T135241
```
(again I'm not sure about the best syntax) then you could add arbitrary information later without having to change the script.

Reply:
I have made it to accept the following syntax:

```
portindex2json.tcl /path/to/PortIndex -info key=value
```

where `path/to/PortIndex` has to be the first argument.

Top level keys will be `"info"` and `"ports"`.

Reply:
> (I'm not sure what the best name would be; `id` is one option, but it could also be called `shasum` or `sha1` or `commit` or whatever.)

Currently, I have let the user decide the key. Would this be a problem?

Reply:
That's perfectly fine.